crates/zoon/src/element/button.rs

```diff
@@ -8,6 +8,60 @@ use std::marker::PhantomData;
 
 make_flags!(Label, OnPress);
 
+/// Html Button for your web page.
+/// You can create a new button by using its constructor and chain styling.
+///
+/// # Examples
+/// ```no_run
+/// use zoon::*;
+/// let button = Button::new().s(Align::center()).s(Padding::all(5));
+/// ```
+/// You can also create your button with specific events as well and update the styling the way you need by using [signals](https://crates.io/crates/futures-signals)
+///
+/// 1 - Create signal for local state management.
+///
+/// 2 - The variable **`hovered`** gets updated when the user hovers the button
+///
+/// 3 - The variable **`hovered_signal`** is the notification send by signal
+/// that will actually determine the background color for the button.
+///
+/// # Example
+/// ```no_run
+/// use zoon::{named_color::*, *};
+/// let (hovered, hovered_signal) = Mutable::new_and_signal(false);
+///
+/// let button = Button::new().s(Background::new()
+///     .on_hovered_change(move |is_hovered| hovered.set_neq(is_hovered))
+///     .color_signal(hovered_signal.map_bool_signal(|| GREEN_7, || GREEN_8)));
+/// ```
+/// It is also possible to with local state and make quick updates.
+///
+/// # Example
+/// ```no_run
+/// use zoon::{*, named_color::*};
+/// use std::{rc::Rc,};
+/// use std::borrow::Cow;
+///     
+/// let click_count = Mutable::new(0);
+///
+/// // Create a title that gets updated when counting changes.
+/// let title = click_count.signal().map(|count| {
+///         if count == 0 {
```

Comment:
Do you think there is a tool or a way to enable example code formatting in Rustfmt?

Reply:
Yes and I did use it actually. I have a non committed  .rustfmt.toml stolen from Seed :laughing: 

I need probably to work out the config

```toml
newline_style = "Unix"
use_field_init_shorthand = true
use_try_shorthand = true
#
#
## nightly
#
normalize_doc_attributes = true
#reorder_impl_items = true
normalize_comments = true
format_code_in_doc_comments = true
wrap_comments = true
#wrap_comments = true
#format_strings = true
#match_block_trailing_comma = true
#trailing_comma = "Always"
#merge_imports = true
#error_on_line_overflow = true
#error_on_unformatted = true
#report_todo = "Always"
#report_fixme = "Always"
#version = "Two"
#max_width=150
```

The code actually does format inside :smile: but incorrectly maybe.

Reply:
I think we could add a `rustfmt.toml` in a next PR